staging/src/k8s.io/apimachinery/pkg/runtime/serializer/allocator.go

```diff
@@ -0,0 +1,21 @@
+package serializer
+
+import "k8s.io/apimachinery/pkg/runtime"
+
+type Allocator struct {
+	buf []byte
+}
+
+var _ runtime.MemoryAllocator = &Allocator{}
```

Comment:
+1

1 buffer will not be enough anyway, because we may be serializing many different objects at once.
Using sync.Pool will allow us to utilize that nicely.

Reply:
good idea, I like it, thanks for pointing it out.

Reply:
I think this should do the trick - https://github.com/kubernetes/kubernetes/pull/108186/commits/6e3c32adf7fe8ab7910f30f7e301255ba57b62e2